clazy/info.yaml

```diff
@@ -0,0 +1,63 @@
+enabled: true
+name: clazy
+command:
+  - clazy-standalone-apply
+arguments:
+  - -checks=level2
+supports_arg_sep: false
+version_cmd: |
+  apt-cache policy clazy | sed '/^ *Installed: \(.*\)$/!d; s//v\1/'
+include:
+  - "**/*.cpp"
+documentation:
+  - https://github.com/KDE/clazy
+metadata:
+  languages:
+    - C++
+  tests:
+    - name: qstring-allocations
+      support:
+        path: CMakeLists.txt
+        contents: |
+          cmake_minimum_required(VERSION 3.16)
+
+          project(helloworld VERSION 1.0.0 LANGUAGES CXX)
+
+          set(CMAKE_CXX_STANDARD 17)
+          set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+          set(CMAKE_AUTOMOC ON)
+
+          find_package(Qt6 REQUIRED COMPONENTS Core)
+
+          add_executable(helloworld
+              clazy-test-1.cpp
+          )
+
+          target_link_libraries(helloworld PRIVATE Qt6::Core)
+      contents: |
+        #include <iostream>
+        #include <QString>
+
+        int main() {
+          QString str = "hello";
+          QString result = str.left(1);
+
+          std::cout << "Original string: " << str.toStdString() << std::endl;
+          std::cout << "Substring from index 5 onwards: " << result.toStdString() << std::endl;
+
+          return 0;
+        }
+      restyled: |
+        #include <iostream>
+        #include <QString>
+
+        int main() {
+          QString str = QStringLiteral("hello");
+          QString result = str.left(1);
```

Comment:
@vadi2, I've finally gotten back to this and I think I'm ready to ship this clazy restyler. However, to get the thing to do _anything_ I had to run it with `-checks=level2`. I would prefer to not do that (and let users decide) but, even though the docs say it runs `level1` by default, and even though this test case is doing something a level1 check (`qstring-left`) should fix (replacing this `left(1)` with `at(0)`), it doesn't happen. Do you have any idea what I'm missing?

Reply:
That sounds good :+1: